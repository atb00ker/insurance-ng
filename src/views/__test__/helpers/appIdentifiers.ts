export enum appTestIds {
  homePhoneNumberInput = "home-phone-number-input",
  sectionLoading = "section-loading",
  featuresPageLogo = "features-page-logo",
  featuresPageButton = "features-page-button"
}

export enum appStringText {
  signUp = "Sign Up"
}
