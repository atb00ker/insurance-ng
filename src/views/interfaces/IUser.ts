export interface IUser {
  id?: string;
  name?: string;
  email?: string;
  jwt: () => Promise<string>;
}

export interface IAuth {
  loginWithRedirect: (options?: any) => Promise<void>;
  logout: (options?: any) => void;
  isAuthenticated?: boolean;
  isReady?: boolean;
  user: IUser;
}
