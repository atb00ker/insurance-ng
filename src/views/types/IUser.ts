export type IUser = {
  id?: string;
  name?: string;
  email?: string;
  jwt: () => Promise<string>;
};

export type IAuth = {
  loginWithRedirect: (options?: any) => Promise<void>;
  logout: (options?: any) => void;
  isAuthenticated?: boolean;
  isReady?: boolean;
  user: IUser;
};

export type IUserProfileScores = {
  title: string;
  explaination: string;
  score: number;
};
