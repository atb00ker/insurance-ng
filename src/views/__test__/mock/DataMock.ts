import { InsuranceTypes } from '../../enums/Insurance';
import { IFIData, IFIInsurance } from '../../types/IFIData';
import { IUser } from '../../types/IUser';

export const userJohnDoe: IUser = {
  id: '12345',
  name: 'John Doe',
  email: 'john@example.com',
  jwt: () => new Promise(() => '12345'),
};

export const insuranceMedicalJohnDoe: IFIInsurance = {
  uuid: '94be9707-d419-404d-a3e2-5be54019fe85',
  type: InsuranceTypes.MEDICAL_PLAN,
  title: 'Medical Plan',
  description: 'Medical Plan Description',
  account_id: 'TESTACCT01',
  score: 0.5,
  clauses: ['Clause 1', 'Clause 2'],
  current_clauses: ['Clause 2', 'Clause 3'],
  current_premium: 1,
  current_cover: 10,
  offer_premium: 1.5,
  offer_cover: 11,
  is_active: true,
  is_claimed: false,
  yoy_deduction_rate: 0.5,
  is_insurance_ng_acct: false
}

export const fiDataJohnDoe: IFIData = {
  status: false,
  data: {
    name: 'John Doe',
    date_of_birth: new Date(2011,10,30),
    pancard: 'TESTPAN01',
    ckyc_compliance: true,
    age_score: 0,
    wealth_score: 0,
    debt_score: 0,
    investment_score: 0,
    phone: '999999999999999',
    shared_data_sources: 0,
    insurance: [insuranceMedicalJohnDoe]
  }
}
