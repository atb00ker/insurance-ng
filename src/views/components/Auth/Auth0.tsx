import React from 'react';
import { useAuth0, Auth0Provider, Auth0ProviderOptions } from '@auth0/auth0-react';
import { AuthContext } from './AuthProvider';
import { IAuth } from '../../types/IUser';

export const Provider = Auth0Provider;
export const ProviderOptions: Auth0ProviderOptions = {
  domain: process.env.AUTH0_DOMAIN || '',
  clientId: process.env.AUTH0_CLIENT_ID || '',
  redirectUri: process.env.AUTH0_REDIRECT_URI || window.location.origin,
  cacheLocation: 'localstorage',
};
const AuthConfigurations: React.FC = ({ children }) => {
  const { isLoading, isAuthenticated, loginWithRedirect, logout, user, getIdTokenClaims } = useAuth0();

  const auth0Logout = () => {
    return logout({ returnTo: process.env.AUTH0_LOGOUT_URI || window.location.origin });
  };

  const getJwt = async (): Promise<string> => {
    if (isAuthenticated) {
      const token = await getIdTokenClaims();
      return token.__raw;
    }
    return new Promise((_, __) => '');
  };

  const auth: IAuth = {
    loginWithRedirect: loginWithRedirect,
    logout: auth0Logout,
    isAuthenticated: isAuthenticated,
    isReady: !isLoading,
    user: {
      id: user?.sub,
      name: user?.name,
      email: user?.email,
      jwt: getJwt,
    },
  };

  // Enable if we want the autologin required feature
  // if (auth.isReady && !auth.isAuthenticated) {
  //   loginWithRedirect();
  //   return <div>Loading...</div>;
  // }
  return (
    <React.Fragment>
      <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>
    </React.Fragment>
  );
};

export { AuthConfigurations };
