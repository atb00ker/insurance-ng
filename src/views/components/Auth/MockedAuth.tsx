import React from 'react';
import { AuthContext } from './AuthProvider';
import { IAuth, IUser } from '../../types/IUser';

const AuthProviderMock: React.FC<{
  children: React.ReactNode;
  isReady: boolean;
  isAuthenticated: boolean;
  user: IUser;
}> = ({ children, isReady, isAuthenticated, user }) => {
  const getJwt = async (): Promise<string> => {
    if (isAuthenticated) {
      return '12345';
    }
    return new Promise(() => '');
  };

  const auth: IAuth = {
    loginWithRedirect: () => {
      return new Promise(() => '');
    },
    logout: () => {},
    isAuthenticated: isAuthenticated,
    isReady: isReady,
    user: {
      id: user.id,
      name: user.name,
      email: user.email,
      jwt: getJwt,
    },
  };

  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
};

export { AuthProviderMock };
