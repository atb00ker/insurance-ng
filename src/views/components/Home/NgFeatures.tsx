import React, { useContext } from 'react';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Image from 'react-bootstrap/Image';
import CompleteLogoImage from '../../assets/icons/complete-logo.jpg';
import './NgFeatures.scss';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import { FeatureDisplayOffWhite } from './FeatureDisplayOffWhite';
import { FeatureDisplayWhite } from './FeatureDisplayWhite';
import EmptyInformationDashboard3 from '../../assets/illustrations/empty-information-dashboard-3.svg';
import SearchPage1 from '../../assets/illustrations/search-page-1.svg';
import SaveMoney1 from '../../assets/illustrations/save-money-1.svg';
import ClauseDiscovery1 from '../../assets/illustrations/clause-discovery-1.svg';
import SteppingToFuture1 from '../../assets/illustrations/stepping-to-future-1.svg';
import { rightArrowInCircle } from '../../helpers/svgIcons';
import { Footer } from '../Common/Footer';
import { AuthContext } from '../Auth/AuthProvider';
import { IAuth } from '../../types/IUser';

const NgFeatures: React.FC = () => {
  const auth: IAuth = useContext(AuthContext);
  return (
    <Container className='overflow-hidden' fluid>
      <Row className='cover-screen'>
        <Col sm='12' className='d-flex-center'>
          <Container className='d-flex-center'>
            <Row className='d-flex-center'>
              <Col sm='12' md='5' className='text-center-md mt-4 max-width-960'>
                <h1>
                  The future of <span className='text-primary'>Insurance Services</span>
                </h1>
                <h5>Insurance management experience tailored to your lifestyle.</h5>
                {auth.isReady && !auth.isAuthenticated && (
                  <Button
                    data-testid='features-page-button'
                    onClick={() => auth.loginWithRedirect()}
                    className='d-inline'
                    variant='primary'>
                    Sign Up {rightArrowInCircle('0 0 16 16')}
                  </Button>
                )}
              </Col>
              <Col
                data-testid='features-page-logo'
                className='text-center'
                xs={{ order: 'first', span: '12' }}
                md={{ order: 'last', span: '7' }}>
                <Image className='d-md-none' src={CompleteLogoImage} width={'90%'} />
                <Image className='d-none d-md-block d-lg-none' src={CompleteLogoImage} width={'400px'} />
                <Image className='d-none d-lg-block' src={CompleteLogoImage} width={'500px'} />
              </Col>
            </Row>
          </Container>
        </Col>
      </Row>
      <FeatureDisplayOffWhite
        image={EmptyInformationDashboard3}
        imageWidth={'450px'}
        title='Variable Premiums'
        description='We offer lower premiums which adjusts to your lifestyle. If you are a low risk taker, why should your premiums be high?'
      />
      <FeatureDisplayWhite
        image={SearchPage1}
        imageWidth={'550px'}
        title='Tailored Dashboard'
        description="Don't waste time searching what you want, we will find it for you and provide an experience created uniquely for you."
      />
      <FeatureDisplayOffWhite
        image={SaveMoney1}
        imageWidth={'550px'}
        title='Find Better Plans'
        description='Do you have an existing insurance, if there exists a better plan in the category, we will suggest it to you.'
      />
      <FeatureDisplayWhite
        image={ClauseDiscovery1}
        imageWidth={'550px'}
        title='Clause Discovery'
        description="No need to go through 18 pages of terms and conditions, clearly see what's applicable for you and for what you are paying."
      />
      <FeatureDisplayOffWhite
        image={SteppingToFuture1}
        imageWidth={'650px'}
        title='See the future'
        description='Your premiums change with time, see the future predictions for upto 5 years, predicted based on your financial status of the past 5 years.'
      />
      <Row className='cover-screen'>
        <Col sm='12' className='d-flex-center'>
          <Container className='d-flex-center'>
            <Row>
              <Col sm='12' className='d-flex-center max-width-960'>
                <h1>Much more...</h1>
              </Col>
              <Col sm='12' className='d-flex-center text-center max-width-960'>
                <h5>
                  No long forms, full control on your data, quick and simple user experience. <br />
                  So, what are you waiting for?
                  <br />
                  {auth.isReady && !auth.isAuthenticated && (
                    <a className='href-no-underline' onClick={() => auth.loginWithRedirect()}>
                      Sign up now and get started {rightArrowInCircle('0 2 20 16')}
                    </a>
                  )}
                </h5>
              </Col>
            </Row>
          </Container>
        </Col>
      </Row>
      <Footer />
    </Container>
  );
};

export { NgFeatures };
