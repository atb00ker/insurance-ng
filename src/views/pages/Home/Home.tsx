import React, { useContext } from 'react';
import Container from 'react-bootstrap/Container';
import { CreateConsent } from '../../components/Home/CreateConsent';
import { IAuth } from '../../types/IUser';
import { AuthContext } from '../../components/Auth/AuthProvider';
import { NgFeatures } from '../../components/Home/NgFeatures';
import { SectionLoader } from '../../components/ContentState/SectionLoader';

const Home: React.FC = () => {
  const auth: IAuth = useContext(AuthContext);

  return (
    <>
      <Container>
        {!auth.isReady && <SectionLoader height='500px' width='100%' />}
        {auth.isReady && auth.isAuthenticated && <CreateConsent auth={auth} />}
      </Container>
      {/* NgFeatures page can be an entire page in itself as well,
        hence it has it's own container. */}
      {auth.isReady && !auth.isAuthenticated && <NgFeatures />}
    </>
  );
};

export { Home };
