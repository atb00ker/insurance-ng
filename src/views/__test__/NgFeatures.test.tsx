/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import 'regenerator-runtime/runtime';
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, act, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { NgFeatures } from '../components/Home/NgFeatures';
import { userJohnDoe } from './mock/DataMock';
import { IUser } from '../types/IUser';
import { AuthProviderMock } from '../components/Auth/MockedAuth';
import { appStringText, appTestIds } from './helpers/appIdentifiers';

const ngFeaturesComponent = (isReady: boolean, isAuthenticated: boolean, user: IUser = userJohnDoe) => (
  <AuthProviderMock isReady={isReady} isAuthenticated={isAuthenticated} user={user}>
    <Router>
      <NgFeatures />
    </Router>
  </AuthProviderMock>
);

describe('NgFeatures page tests', () => {
  // beforeEach(() => {});
  afterEach(() => cleanup);

  it('renders without crashing', () => {
    const root = document.createElement('div');
    ReactDOM.render(ngFeaturesComponent(true, false), root);
    ReactDOM.render(ngFeaturesComponent(true, true), root);
  });

  it('test sign up button not exists after auth', () => {
    let page: RenderResult | any;
    act(() => {
      page = render(ngFeaturesComponent(true, true));
    });
    expect(page.getByTestId(appTestIds.featuresPageLogo)).toBeInTheDocument();
    expect(page.queryByText(appStringText.signUp)).not.toBeInTheDocument();
  });

  it('test sign up button exists before auth', () => {
    let page: RenderResult | any;
    act(() => {
      page = render(ngFeaturesComponent(true, false));
    });
    expect(page.getByTestId(appTestIds.featuresPageLogo)).toBeInTheDocument();
    expect(page.getByTestId(appTestIds.featuresPageButton)).toBeInTheDocument();
  });

  it('ngfeature snapshot structure', () => {
    const snapTree = renderer.create(ngFeaturesComponent(true, false));
    expect(snapTree.toJSON()).toMatchSnapshot();
  });
});
