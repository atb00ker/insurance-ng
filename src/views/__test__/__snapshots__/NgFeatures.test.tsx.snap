// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgFeatures page tests ngfeature snapshot structure 1`] = `
<div
  className="overflow-hidden container-fluid"
>
  <div
    className="cover-screen row"
  >
    <div
      className="d-flex-center col-sm-12"
    >
      <div
        className="d-flex-center container"
      >
        <div
          className="d-flex-center row"
        >
          <div
            className="text-center-md mt-4 max-width-960 col-md-5 col-sm-12"
          >
            <h1>
              The future of 
              <span
                className="text-primary"
              >
                Insurance Services
              </span>
            </h1>
            <h5>
              Insurance management experience tailored to your lifestyle.
            </h5>
            <button
              className="d-inline btn btn-primary"
              data-testid="features-page-button"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Sign Up 
              <svg
                className="bi bi-arrow-right-circle-fill"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                />
              </svg>
            </button>
          </div>
          <div
            className="text-center col-md-7 col-12 order-md-last order-first"
            data-testid="features-page-logo"
          >
            <img
              className="d-md-none"
              src={Object {}}
              width="90%"
            />
            <img
              className="d-none d-md-block d-lg-none"
              src={Object {}}
              width="400px"
            />
            <img
              className="d-none d-lg-block"
              src={Object {}}
              width="500px"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen bg-off-white row"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="h-100 max-width-960 container"
      >
        <div
          className="h-100 row"
        >
          <div
            className="cover-screen-flex col-md-8 col-12"
          >
            <img
              alt="Variable Premiums"
              className="d-none d-md-block cover-screen-flex"
              src={Object {}}
              width="450px"
            />
            <img
              alt="Variable Premiums"
              className="d-md-none cover-screen-flex"
              src={Object {}}
              width="100%"
            />
          </div>
          <div
            className="col-md-4 col-12"
          >
            <div
              className="cover-screen-flex"
            >
              <div
                className="row"
              >
                <div
                  className="col-12"
                >
                  <h2
                    className="text-primary"
                  >
                    Variable Premiums
                  </h2>
                </div>
                <div
                  className="col-12"
                >
                  We offer lower premiums which adjusts to your lifestyle. If you are a low risk taker, why should your premiums be high?
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen row"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="h-100 max-width-960 container"
      >
        <div
          className="h-100 row"
        >
          <div
            className="col-md-4 col-12"
          >
            <div
              className="cover-screen-flex"
            >
              <div
                className="row"
              >
                <div
                  className="col-12"
                >
                  <h2
                    className="text-primary"
                  >
                    Tailored Dashboard
                  </h2>
                </div>
                <div
                  className="col-12"
                >
                  Don't waste time searching what you want, we will find it for you and provide an experience created uniquely for you.
                </div>
              </div>
            </div>
          </div>
          <div
            className="cover-screen-flex col-md-8 col-12 order-md-last order-first"
          >
            <img
              alt="Tailored Dashboard"
              className="d-none d-md-block cover-screen-flex"
              src={Object {}}
              width="550px"
            />
            <img
              alt="Tailored Dashboard"
              className="d-md-none cover-screen-flex"
              src={Object {}}
              width="100%"
            />
             
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen bg-off-white row"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="h-100 max-width-960 container"
      >
        <div
          className="h-100 row"
        >
          <div
            className="cover-screen-flex col-md-8 col-12"
          >
            <img
              alt="Find Better Plans"
              className="d-none d-md-block cover-screen-flex"
              src={Object {}}
              width="550px"
            />
            <img
              alt="Find Better Plans"
              className="d-md-none cover-screen-flex"
              src={Object {}}
              width="100%"
            />
          </div>
          <div
            className="col-md-4 col-12"
          >
            <div
              className="cover-screen-flex"
            >
              <div
                className="row"
              >
                <div
                  className="col-12"
                >
                  <h2
                    className="text-primary"
                  >
                    Find Better Plans
                  </h2>
                </div>
                <div
                  className="col-12"
                >
                  Do you have an existing insurance, if there exists a better plan in the category, we will suggest it to you.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen row"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="h-100 max-width-960 container"
      >
        <div
          className="h-100 row"
        >
          <div
            className="col-md-4 col-12"
          >
            <div
              className="cover-screen-flex"
            >
              <div
                className="row"
              >
                <div
                  className="col-12"
                >
                  <h2
                    className="text-primary"
                  >
                    Clause Discovery
                  </h2>
                </div>
                <div
                  className="col-12"
                >
                  No need to go through 18 pages of terms and conditions, clearly see what's applicable for you and for what you are paying.
                </div>
              </div>
            </div>
          </div>
          <div
            className="cover-screen-flex col-md-8 col-12 order-md-last order-first"
          >
            <img
              alt="Clause Discovery"
              className="d-none d-md-block cover-screen-flex"
              src={Object {}}
              width="550px"
            />
            <img
              alt="Clause Discovery"
              className="d-md-none cover-screen-flex"
              src={Object {}}
              width="100%"
            />
             
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen bg-off-white row"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="h-100 max-width-960 container"
      >
        <div
          className="h-100 row"
        >
          <div
            className="cover-screen-flex col-md-8 col-12"
          >
            <img
              alt="See the future"
              className="d-none d-md-block cover-screen-flex"
              src={Object {}}
              width="650px"
            />
            <img
              alt="See the future"
              className="d-md-none cover-screen-flex"
              src={Object {}}
              width="100%"
            />
          </div>
          <div
            className="col-md-4 col-12"
          >
            <div
              className="cover-screen-flex"
            >
              <div
                className="row"
              >
                <div
                  className="col-12"
                >
                  <h2
                    className="text-primary"
                  >
                    See the future
                  </h2>
                </div>
                <div
                  className="col-12"
                >
                  Your premiums change with time, see the future predictions for upto 5 years, predicted based on your financial status of the past 5 years.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cover-screen row"
  >
    <div
      className="d-flex-center col-sm-12"
    >
      <div
        className="d-flex-center container"
      >
        <div
          className="row"
        >
          <div
            className="d-flex-center max-width-960 col-sm-12"
          >
            <h1>
              Much more...
            </h1>
          </div>
          <div
            className="d-flex-center text-center max-width-960 col-sm-12"
          >
            <h5>
              No long forms, full control on your data, quick and simple user experience. 
              <br />
              So, what are you waiting for?
              <br />
              <a
                className="href-no-underline"
                onClick={[Function]}
              >
                Sign up now and get started 
                <svg
                  className="bi bi-arrow-right-circle-fill"
                  fill="currentColor"
                  height="16"
                  viewBox="0 2 20 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                  />
                </svg>
              </a>
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="mb-3 text-center col-sm-12"
    >
      Made with 
      <span
        className="text-danger"
      >
        <svg
          className="bi bi-arrow-right-circle-fill"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
            fillRule="evenodd"
          />
        </svg>
      </span>
       by
       
      <a
        className="href-no-underline"
        href="https://github.com/atb00ker"
      >
        @atb00ker
      </a>
    </div>
  </div>
</div>
`;
